tests/suites/test_suite_ecdh.function

```diff
@@ -131,12 +131,14 @@ void ecdh_exchange( int id )
     const unsigned char *vbuf;
     size_t len;
     rnd_pseudo_info rnd_info;
+    unsigned char res_buf[1000];
+    size_t res_len;
 
     mbedtls_ecdh_init( &srv );
     mbedtls_ecdh_init( &cli );
     memset( &rnd_info, 0x00, sizeof( rnd_pseudo_info ) );
 
-    TEST_ASSERT( mbedtls_ecp_group_load( &srv.grp, id ) == 0 );
+    TEST_ASSERT( mbedtls_ecdh_setup( &srv, id ) == 0 );
```

Comment:
Should we keep a test with `mbedtls_ecp_group_load()` as well to ensure we don't break the API?

Reply:
We are using `mbedtls_ecp_group_load()` in the other two test functions in this suite, I think that should be enough to ensure that we don't break the API.

Reply:
OK

Reply:
Follow-up questions: will those two other tests still work when the new configuration is used with an alternate implementation of a particular group? In general, which of our tests will be available for alternative group implementers to check their work with, and how many depend on the group being Mbed TLS implemented (i.e. using `MBEDTLS_ECDH_VARIANT_MBEDTLS_2_0`)?

Reply:
Ok, I think I misread the tests, the other two tests don't use these contexts, they exercise a lower level API.

I'll add a test to exercise the legacy workflow.